addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,151 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             $carriers.filter(':checked').click();
         }
 
-        // Asynchronously retrieve every carrier price
+        self._getCurrentLocation();
+
         _.each($carriers, function (carrierInput, k) {
             self._showLoading($(carrierInput));
             self._rpc({
                 route: '/shop/carrier_rate_shipment',
                 params: {
                     'carrier_id': carrierInput.value,
                 },
-            }).then(self._handleCarrierUpdateResultBadge.bind(self));
+            }).then(self._handleCarrierUpdateResult.bind(self));
         });
-
+        
         return this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @private
+     */
+    _getCurrentLocation: function () {
+        const self=this;
+        self._rpc({
+            route: "/shop/access_point/get",
+        }).then(function (data) {
+            const order_locations = document.getElementsByClassName("o_order_location");
+            for (const order_loc of order_locations) {
+                const show_loc = order_loc.parentElement.nextElementSibling;
+                const delivery_type = order_loc.parentElement.parentElement.parentNode.children[0].attributes[8].nodeValue;
+                if (!show_loc)
+                    continue;
+                if (data[delivery_type+'_access_point']) {
+                    order_loc.innerText = data[delivery_type+'_access_point']
+                    order_loc.parentElement.classList.remove("d-none");
+                    show_loc.classList.add("d-none");
+                    break;
+                } else {
+                    order_loc.parentElement.classList.add("d-none");
+                    show_loc.classList.remove("d-none");
+                }
+            }
+        });
+    },
```

Comment:
Don't hesitate to use async functions, it's more modern and easier to work with.
```suggestion
    _getCurrentLocation: async function () {
        const self=this;
        const data = await self._rpc({...});
        const order_locations = ...
    },
```